src/mrnet/core/reactions.py

```diff
@@ -1686,135 +1800,136 @@ def generate(
+        and rxn_type_B is the reverse.
 
-        rxn_type_A = "Concerted"
-        rxn_type_B = "Concerted"
+        Returns: None
+        """
 
-        reaction_type = {
-            "class": "ConcertedReaction",
-            "rxn_type_A": rxn_type_A,
-            "rxn_type_B": rxn_type_B,
-        }
-        return reaction_type
+        self.rxn_type_A = "Concerted"
+        self.rxn_type_B = "Concerted"
+        return
 
-    def free_energy(self, temperature=298.15) -> Mapping_Energy_Dict:
+    def free_energy(self, temperature=298.15):
         """
         A method to determine the free energy of the concerted reaction
+        Sets free_energy_A and free_energy_B,
+        where free_energy_A is the primary type of the reaction based on
+        the reactant and product of the ConcertedReaction
+        object, and the backwards of this reaction would be free_energy_B.
         Args:
-           :return dictionary of the form {"free_energy_A": energy_A,
-                                           "free_energy_B": energy_B}
-           where free_energy_A is the primary type of the reaction based on
-           the reactant and product of the ConcertedReaction
-           object, and the backwards of this reaction would be free_energy_B.
+            temperature:
+        Returns:
+            None
         """
 
+        set_base = False
+        if temperature is None or temperature == 298.15:
+            if self.base_free_energy_A is not None and self.base_free_energy_B is not None:
+                self.free_energy_A = self.base_free_energy_A
+                self.free_energy_B = self.base_free_energy_B
+            else:
+                set_base = True
+
         if self.electron_free_energy is None:
             electron_free = 0.0
         else:
             electron_free = self.electron_free_energy
 
-        cond_rct = all(reactant.get_free_energy() is not None for reactant in self.reactants)
-        cond_pro = all(product.get_free_energy() is not None for product in self.products)
-        if cond_rct and cond_pro:
-            reactant_charge = np.sum([item.charge for item in self.reactants])
-            product_charge = np.sum([item.charge for item in self.products])
-            reactant_free_energy = np.sum(
-                [item.get_free_energy(temp=temperature) for item in self.reactants]
+        rct_free_energies = [
+            mol_free_energy(
+                self.reactant_energy[i],
+                self.reactant_enthalpy[i],
+                self.reactant_entropy[i],
+                temp=temperature,
             )
-            product_free_energy = np.sum(
-                [item.get_free_energy(temp=temperature) for item in self.products]
+            for i in range(len(self.reactant_eids))
+        ]
+        pro_free_energies = [
+            mol_free_energy(
+                self.product_energy[i],
+                self.product_enthalpy[i],
+                self.product_entropy[i],
```

Comment:
This is what I'm talking about. All classes should have a self.product_energy that's a list. Even if it's a list of length 1 always (those special classes can also have a separate property that's just equal to self.product_energy[0].